src/stirling/bpf_tools/bcc_wrapper.h

```diff
@@ -227,27 +219,84 @@ class BCCWrapper {
    *                   Default is 0, because if nothing is ready, then we want to go back to sleep
    *                   and catch new events in the next iteration.
    */
-  void PollPerfBuffers(int timeout_ms = 0);
-  void PollPerfBuffer(std::string_view perf_buffer_name, int timeout_ms);
+  virtual void PollPerfBuffers(int timeout_ms = 0) = 0;
+  virtual void PollPerfBuffer(std::string_view perf_buffer_name, int timeout_ms) = 0;
 
   /**
    * Detaches all probes, and closes all perf buffers that are open.
    */
-  void Close();
+  virtual void Close() = 0;
 
   // These are static counters of attached/open probes across all instances.
   // It is meant for verification that we have cleaned-up all resources in tests.
   static size_t num_attached_probes() { return num_attached_kprobes_ + num_attached_uprobes_; }
   static size_t num_open_perf_buffers() { return num_open_perf_buffers_; }
   static size_t num_attached_perf_events() { return num_attached_perf_events_; }
 
+ protected:
+  FRIEND_TEST(BCCWrapperTest, DetachUProbe);
+  virtual Status ClosePerfBuffer(const PerfBufferSpec& perf_buffer) = 0;
+
+  // These are static counters across all instances, because:
+  // 1) We want to ensure we have cleaned all BPF resources up across *all* instances (no leaks).
+  // 2) It is for verification only, and it doesn't make sense to create accessors from stirling to
+  // here.
+  inline static size_t num_attached_kprobes_;
+  inline static size_t num_attached_uprobes_;
+  inline static size_t num_attached_tracepoints_;
+  inline static size_t num_open_perf_buffers_;
+  inline static size_t num_attached_perf_events_;
```

Comment:
It seems these were duplicated (they already exist [here](https://github.com/pixie-io/pixie/pull/1660/files#diff-3b928f7d097d85c29260dcf7bbd52ec73f85700814c45ec4c49e92c73734966dR337)).

Reply:
Thank you! ... this slipped in somehow. Removed it.